pandas/core/series.py

```diff
@@ -581,6 +585,17 @@ def __init__(
         self.name = name
         self._set_axis(0, index)
 
+        if original_dtype is None and is_pandas_object and data_dtype == np.object_:
+            if self.dtype != data_dtype:
+                warnings.warn(
+                    "Dtype inference on a pandas object "
+                    "(Series, Index, ExtensionArray) is deprecated. The Index "
+                    "constructor will keep the original dtype in the future. "
+                    "Call ``infer_objects on the result",
+                    FutureWarning,
+                    stacklevel=2,
```

Comment:
Does `find_stack_level` return the incorrect result here?

Reply:
This is nice for internal debugging, but yeah I can use find_stack_level as well

Reply:
updated